snap/patches/nova/0001-remotefs-http-driver.patch

```diff
@@ -0,0 +1,246 @@
+diff --git a/nova/conf/libvirt.py b/nova/conf/libvirt.py
+index 0746d4040a..099dfd5762 100644
+--- a/nova/conf/libvirt.py
++++ b/nova/conf/libvirt.py
+@@ -1384,7 +1384,7 @@ and ``gid`` must be specified.
+ libvirt_remotefs_opts = [
+     cfg.StrOpt('remote_filesystem_transport',
+                default='ssh',
+-               choices=('ssh', 'rsync'),
++               choices=('ssh', 'rsync', 'http'),
+                help="""
+ libvirt's transport method for remote file operations.
+ 
+@@ -1395,7 +1395,29 @@ compute nodes, other method must be used for:
+ * creating file on remote host
+ * removing file from remote host
+ * copying file to remote host
+-""")
++"""),
++    cfg.IntOpt('file_server_port',
++               default=10099,
++               min=1,
++               help='TCP port of the remote file server used by the http '
++                    'RemoteFilesystem driver.'),
++    cfg.IntOpt('file_server_read_timeout',
++               default=900,
++               min=1,
++               help='Read timeout in seconds for HTTP operations in the http '
++                    'RemoteFilesystem driver.'),
++    cfg.StrOpt('file_server_ca_file',
++               default='',
++               help='Path to a CA bundle file to verify the HTTPS file '
++                    'server. If set, verification is enabled using this file.'),
++    cfg.StrOpt('file_server_client_cert_file',
++               default='',
++               help='Path to client certificate file for mutual TLS to the '
++                    'file server.'),
++    cfg.StrOpt('file_server_client_key_file',
++               default='',
++               help='Path to client private key file for mutual TLS to the '
++                     'file server.'),
+ ]
+ 
+ libvirt_volume_vzstorage_opts = [
+diff --git a/nova/virt/libvirt/volume/remotefs.py b/nova/virt/libvirt/volume/remotefs.py
+index 8ddcaa1fcc..8096c77903 100644
+--- a/nova/virt/libvirt/volume/remotefs.py
++++ b/nova/virt/libvirt/volume/remotefs.py
+@@ -24,6 +24,8 @@ from oslo_utils import importutils
+ import nova.conf
+ import nova.privsep.fs
+ from nova import utils
++from nova.virt.libvirt.volume.remotefs_http import HttpDriver as _HttpDriverImpl
++HttpDriver = _HttpDriverImpl
+ 
+ LOG = logging.getLogger(__name__)
+ 
+diff --git a/nova/virt/libvirt/volume/remotefs_http.py b/nova/virt/libvirt/volume/remotefs_http.py
+new file mode 100644
+index 0000000000..ed983e2963
+--- /dev/null
++++ b/nova/virt/libvirt/volume/remotefs_http.py
+@@ -0,0 +1,174 @@
++import os
++import ssl
++import tempfile
++from typing import Tuple
++
++from oslo_log import log as logging
++
++import nova.conf
++
++
++LOG = logging.getLogger(__name__)
++CONF = nova.conf.CONF
++
++LOG.debug("[webdav] remotefs_http module loaded")
++
++
++def _debug_log_path(label: str, path: str) -> str:
++    LOG.debug("[webdav] %s path=%s", label, path)
++    return path
```

Comment:
No need to return path if this function is for debug log purpose.
I am not sure if we really need separate function for this, there are hardly 5 places this function is used and there is no logic in this function.